src/librustc_driver/lib.rs

```diff
@@ -151,6 +157,44 @@ pub fn run<F>(run_compiler: F) -> isize
     0
 }
 
+#[cfg(not(feature="llvm"))]
+pub struct NoLLvmMetadataLoader;
+
+#[cfg(not(feature="llvm"))]
+type LlvmMetadataLoader = NoLLvmMetadataLoader;
+#[cfg(feature="llvm")]
+type LLvmMetadataLoader = rustc_trans::LlvmMetadataLoader;
+
+#[cfg(not(feature="llvm"))]
+impl MetadataLoader for NoLLvmMetadataLoader {
+    fn get_rlib_metadata(&self, _: &Target, filename: &Path) -> Result<ErasedBoxRef<[u8]>, String> {
```

Comment:
Then it should be depended on by `rustc_metadata` and `get_rlib_metadata` shouldn't exist which means `loader: Box<MetadataLoader>` in `rustc_metadata` becomes just:
```rust
get_dylib_metadata: fn(&Target,  &Path) -> Result<ErasedBoxRef<[u8]>, String>
```
When not available, one can just pass in `|_, _| Err(String::from("unsupported"))`.